packages/sns/src/converters/governance.converters.ts

```diff
@@ -264,3 +284,206 @@ export const toListProposalRequest = ({
+  canister_upgrade_arg: fromNullable(params.canister_upgrade_arg),
+});
+
+const convertTransferSnsTreasuryFunds = (
+  params: TransferSnsTreasuryFundsCandid
+): TransferSnsTreasuryFunds => ({
+  from_treasury: params.from_treasury,
+  to_principal: fromNullable(params.to_principal),
+  to_subaccount: fromNullable(params.to_subaccount),
+  memo: fromNullable(params.memo),
+  amount_e8s: params.amount_e8s,
+});
+
+const convertGenericNervousSystemFunction = (
+  params: GenericNervousSystemFunctionCandid
+): GenericNervousSystemFunction => ({
+  validator_canister_id: fromNullable(params.validator_canister_id),
+  target_canister_id: fromNullable(params.target_canister_id),
+  validator_method_name: fromNullable(params.validator_method_name),
+  target_method_name: fromNullable(params.target_method_name),
+});
+
+const convertFunctionType = (
+  params: FunctionTypeCandid | undefined
+): FunctionType | undefined => {
+  if (params === undefined) {
+    return undefined;
+  }
+
+  if ("NativeNervousSystemFunction" in params) {
+    return { NativeNervousSystemFunction: params.NativeNervousSystemFunction };
+  }
+
+  if ("GenericNervousSystemFunction" in params) {
+    return {
+      GenericNervousSystemFunction: convertGenericNervousSystemFunction(
+        params.GenericNervousSystemFunction
+      ),
+    };
+  }
+
+  throw new Error(`Unknown FunctionType ${JSON.stringify(params)}`);
+};
+
+const convertNervousSystemFunction = (
+  params: NervousSystemFunctionCandid
+): NervousSystemFunction => ({
+  id: params.id,
+  name: params.name,
+  description: fromNullable(params.description),
+  function_type: convertFunctionType(fromNullable(params.function_type)),
+});
+
+const convertVotingRewardsParameters = (
+  params: VotingRewardsParametersCandid | undefined
+): VotingRewardsParameters | undefined =>
+  params && {
+    final_reward_rate_basis_points: fromNullable(
+      params.final_reward_rate_basis_points
+    ),
+    initial_reward_rate_basis_points: fromNullable(
+      params.initial_reward_rate_basis_points
+    ),
+    reward_rate_transition_duration_seconds: fromNullable(
+      params.reward_rate_transition_duration_seconds
+    ),
+    round_duration_seconds: fromNullable(params.round_duration_seconds),
+  };
+
+const convertNervousSystemParams = (
+  params: NervousSystemParametersCandid
+): NervousSystemParameters => ({
+  default_followees: fromNullable(params.default_followees),
+  max_dissolve_delay_seconds: fromNullable(params.max_dissolve_delay_seconds),
+  max_dissolve_delay_bonus_percentage: fromNullable(
+    params.max_dissolve_delay_bonus_percentage
+  ),
+  max_followees_per_function: fromNullable(params.max_followees_per_function),
+  neuron_claimer_permissions: fromNullable(params.neuron_claimer_permissions),
+  neuron_minimum_stake_e8s: fromNullable(params.neuron_minimum_stake_e8s),
+  max_neuron_age_for_age_bonus: fromNullable(
+    params.max_neuron_age_for_age_bonus
```

Comment:
Why do you think you couldn't make a test fail?

If this wrote `neuron_minimum_stake_e8s` because you copy/pasted and forgot to change it, the compiler wouldn't complain but a test would hopefully fail, right?

Reply:
Yeah, I was working on it and I could easily make a test fail.

But not sure how much value the tests add anyway.